test/functional/wallet_miniscript.py

```diff
@@ -0,0 +1,93 @@
+    # Liquid-like federated pegin with emergency recovery keys
+    "or_i(and_b(pk(029ffbe722b147f3035c87cb1c60b9a5947dd49c774cc31e94773478711a929ac0),a:and_b(pk(025f05815e3a1a8a83bfbb03ce016c9a2ee31066b98f567f6227df1d76ec4bd143),a:and_b(pk(025625f41e4a065efc06d5019cbbd56fe8c07595af1231e7cbc03fafb87ebb71ec),a:and_b(pk(02a27c8b850a00f67da3499b60562673dcf5fdfb82b7e17652a7ac54416812aefd),s:pk(03e618ec5f384d6e19ca9ebdb8e2119e5bef978285076828ce054e55c4daf473e2))))),and_v(v:thresh(2,pkh(tpubD6NzVbkrYhZ4YK67cd5fDe4fBVmGB2waTDrAt1q4ey9HPq9veHjWkw3VpbaCHCcWozjkhgAkWpFrxuPMUrmXVrLHMfEJ9auoZA6AS1g3grC/*),a:pkh(033841045a531e1adf9910a6ec279589a90b3b8a904ee64ffd692bd08a8996c1aa),a:pkh(02aebf2d10b040eb936a6f02f44ee82f8b34f5c1ccb20ff3949c2b28206b7c1068)),older(4209713)))",
+]
+
+
+class WalletMiniscriptTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+        self.skip_if_no_sqlite()
+
+    def watchonly_test(self, ms):
+        self.log.info(f"Importing Miniscript '{ms}'")
+        desc = descsum_create(f"wsh({ms})")
+        assert self.ms_wo_wallet.importdescriptors(
+            [
+                {
+                    "desc": desc,
+                    "active": True,
+                    "range": 2,
+                    "next_index": 0,
+                    "timestamp": "now",
+                }
+            ]
+        )[0]["success"]
+
+        self.log.info("Testing we derive new addresses for it")
+        assert_equal(
+            self.ms_wo_wallet.getnewaddress(), self.funder.deriveaddresses(desc, 0)[0]
+        )
+        assert_equal(
+            self.ms_wo_wallet.getnewaddress(), self.funder.deriveaddresses(desc, 1)[1]
+        )
+
+        self.log.info("Testing we detect funds sent to one of them")
+        addr = self.ms_wo_wallet.getnewaddress()
+        txid = self.funder.sendtoaddress(addr, 0.01)
+        self.wait_until(
+            lambda: len(self.ms_wo_wallet.listunspent(minconf=0, addresses=[addr])) == 1
+        )
+        utxo = self.ms_wo_wallet.listunspent(minconf=0, addresses=[addr])[0]
+        assert utxo["txid"] == txid and not utxo["solvable"]  # No satisfaction logic (yet)
+
+    def run_test(self):
+        self.log.info("Making a descriptor wallet")
+        self.funder = self.nodes[0].get_wallet_rpc(self.default_wallet_name)
+        self.nodes[0].createwallet(
+            wallet_name="ms_wo", descriptors=True, disable_private_keys=True
+        )
+        self.ms_wo_wallet = self.nodes[0].get_wallet_rpc("ms_wo")
+
+        # Sanity check we wouldn't let an insane Miniscript descriptor in
```

Comment:
need to define insane(I don't know what that means)

Reply:
It was discussed in the OP of the Miniscript PR, #24147, and is defined in `src/script/miniscript.h`:
https://github.com/bitcoin/bitcoin/blob/062b9db0ccb6af8bfbaa2b29132408cda9991b40/src/script/miniscript.h#L842-L846